### ISSUE TITLE
Image is rasterised off-centre/rearranged

### ISSUE THREAD

**Issue by kortschak** [tdewolff/canvas#299]
When the program [here](https://go.dev/play/p/WG_Ic9-BOAM) is run, it renders the svg displaced to the right in the image with the relationships between the elements altered.

![logo](https://github.com/tdewolff/canvas/assets/275221/13832125-90c6-4675-a35d-43da9137d347)

The expected image is instead like this.

![logo](https://github.com/tdewolff/canvas/assets/275221/d722dea7-7dbf-43b1-9715-f60ecab7f236)

My guess is that this has to do with the rotations used in the SVG.

**Comment by kortschak** [comment]
The issue is that `canvas.Matrix.RotateAbout` is returning a matrix that is used relative to the coordinate system for each of the rects, rather than relative to the g.

**Comment by tdewolff** [comment]
Hi Dan, thanks for the bug report. The issue is indeed with the affine transformations that happen in `canvas`. Unfortunately, the current implementation is needlessly complex and I'm unsure what the added value is. Specifically, there is a distinction between coordinate transformations and object transformations. The idea being that you might want to use a different coordinate system for coordinates (for example, place markers in a plot using the axes' units) while not deforming the markers themselves. This is where `CoordView` and `View` came about.

I've revised the implementation and reverted to an earlier implementation of CoordView vs View, where View is the complete affine transformation matrix the same as the `transform` attribute in SVG. The CoordView only transforms coordinates (before View).

I've also implemented rounded rectangles and fixed some problems I encountered. Let me know if this works well for you.
